project/Dependencies.scala

```diff
@@ -36,4 +42,13 @@ object Dependencies {
     "com.chuusai" %% "shapeless" % "2.3.3",
     scalacheck
   )
+
+  private def supportedScalaVersions(scalaVersion: String): List[String] = {
+    val split = scalaVersion.split('.')
+    val binaryVersion = split.take(2).mkString(".")
+    val compilerVersion = Try(split.last.toInt).toOption
+    val supportedVersions =
+      compilerVersion.map(version => List.range(version - 2, version + 1).filter(_ >= 0)).getOrElse(Nil)
+    supportedVersions.map(v => s"$binaryVersion.$v")
+  }
```

Comment:
I think this should be called after what it does, not after what it is used for
```suggestion
  private def latestVersions(scalaVersion: String): List[String] = {
    val split = scalaVersion.split('.')
    val binaryVersion = split.take(2).mkString(".")
    val compilerVersion = Try(split.last.toInt).toOption
    val latestPatchVersions =
      compilerVersion.map(version => List.range(version - 2, version + 1).filter(_ >= 0)).getOrElse(Nil)
    latestPatchVersions.map(v => s"$binaryVersion.$v")
  }
```